### PR TITLE
[SK-7860] Adds react package for using Flagr

### PR DESCRIPTION
**Pull request by benmardana**
Adds a small reusable package for using the [flagr-feature-typescript](https://www.npmjs.com/package/flagr-feature-typescript) package

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,52 @@
...
+const APP_TAGS = ['local'];
+
+const MyApp = () => (
+  <FlagrContextProvider flagrUrl={SOME_URL} tags={APP_TAGS}>
+    <MyComponent />
+  </FlagrContextProvider>
+);
+
+const MyComponent = () => {
+  const { featureIsOn } = useFlagr();
```
Comment (Toofifty) [inline README.md]: where were u wen `hasFeature` die

File: lib/index.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  flagrUrl: string;
+  tags: [string, ...string[]];
+}
+
+export const FlagrContextProvider = ({
+  children,
+  flagrUrl,
+  tags,
+}: FlagrContextProviderProps) => {
+  const [value, setValue] = useState<FlagrContextType>(null!);
```
Comment (Toofifty) [inline lib/index.tsx]: Won't there be problems accessing properties on this before the `batchEvaluation` is complete?

Comment (benmardana) [reply lib/index.tsx]: oops forgot to put in a better default

Comment (benmardana) [reply lib/index.tsx]: @Toofifty updated

File: README.md (hunk 1 above)
Comment (benmardana) [reply README.md]: 🤔 i dun get it 🤕

Comment (Toofifty) [reply README.md]: I was at house eating dorito when phone ring

"`hasFeature` is kil"

"no"

Comment (benmardana) [reply README.md]: lmao. RIP in peace `hasFeature`

File: README.md [hunk 3]
```diff
@@ -1 +1,52 @@
...
+## Installation
+
+\*Until [flagr-feature-typescript](https://github.com/getsidekicker/flagr-feature-typescript) is published, it will need to be locally installed and linked manually before running install
+
+```bash
+# In /flagr-feature-typescript/
+npm link
+# In this project
+npm link flagr-feature-typescript --save
+npm install
```
Comment (tristankenney) [inline README.md:36]: **suggestion:** you can kill this – I'm gonna publish the other package today

File: lib/index.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import React, { createContext, useContext, useEffect, useState } from 'react';
+import { createEvaluator, FlagCallbacks } from 'flagr-feature-typescript';
+
+type FlagrContextType = {
+  evaluateFeature?: <T>(flag: string, callbacks: FlagCallbacks<T>) => T;
+  featureIsOn?: (flag: string) => boolean;
```
Comment (tristankenney) [inline lib/index.tsx]: **question:** is this type intentionally different from `match`?

```suggestion
  featureIsOn?: (flag: string, matchVariant: string) => boolean;
```

File: package.json [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  "version": "0.0.1",
+  "description": "react module for using flagr-feature-typescript",
+  "keywords": [
+    "flagr"
+  ],
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/getsidekicker/flagr-feature-react"
+  },
+  "license": "UNLICENSED",
```
Comment (tristankenney) [inline package.json]: **thought:** I've MIT'd all of the others

File: lib/index.tsx (hunk 4 above)
Comment (benmardana) [reply lib/index.tsx]: I was providing a shorthand for the default behaviour but on second look it isn't worth obfuscating. Updated to pass both fns straight through

File: package.json (hunk 5 above)
Comment (benmardana) [reply package.json]: `¯\_(ツ)_/¯` not fussed - just added so npm quiets down

File: README.md [hunk 6]
```diff
@@ -1 +1,40 @@
...
+## Installation
+
+```bash
+npm install
+```
+
+## Development
+
+```bash
+npm run dev
```
Comment (chintogtokh) [inline README.md:42]: **question** Do we have a preference for `yarn` over `npm`?

Comment (benmardana) [reply README.md:42]: I don't think we have a global preference? I think yarn is only used in the frontend projects as npm was too slow at the time they were set up? Here I went with npm (v7) as there are no features in yarn that we need so one less dependency seemed like a good thing™️. In any case, if performance is a consideration, I think npm 7 has resolved many of those issues - https://github.com/npm/benchmarks

File: package.json (hunk 5 above)
Comment (tristankenney) [reply package.json]: Cool – let's go with MIT then

File: README.md [hunk 7]
```diff
@@ -1 +1,44 @@
...
+const APP_TAGS = ['local'];
+
+const MyApp = () => (
+  <FlagrContextProvider flagrUrl={SOME_URL} tags={APP_TAGS}>
+    <MyComponent />
+  </FlagrContextProvider>
+);
+
+const MyComponent = () => {
+  const { featureIsOn } = useFlagr();
```
Comment (tristankenney) [inline README.md]:
```suggestion
  const { match } = useFlagr();
```

File: README.md [hunk 8]
```diff
@@ -1 +1,44 @@
...
+
+  return featureIsOn?.('someFeature') ? 'new hotness' : 'old and busted';
```
Comment (tristankenney) [inline README.md]:
```suggestion
  return match?.('someFeature') ? 'new hotness' : 'old and busted';
```

File: lib/index.tsx [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+  const [value, setValue] = useState<FlagrContextType>({});
+
+  useEffect(() => {
+    (async () => {
+      const Evaluator = createEvaluator({
+        flagrUrl,
+      });
+
+      const { cachedEvaluate, cachedMatch } = await Evaluator.batchEvaluation({
+        context: {},
```
Comment (tristankenney) [inline lib/index.tsx]: **thought:** we're going to want to be able to specify this in the not too distant future. Non-plussed as to whether it's promoted to prop now or laterz

**Review by tristankenney — APPROVED** [review]
SEND EEET

File: lib/index.tsx (hunk 9 above)
Comment (benmardana) [reply lib/index.tsx]: Added now

File: package.json (hunk 5 above)
Comment (benmardana) [reply package.json]: Donezo

File: README.md (hunk 7 above)
Comment (benmardana) [reply README.md]: Good catch - updated

File: README.md (hunk 8 above)
Comment (benmardana) [reply README.md]: Good catch - updated

**Review by Toofifty — APPROVED** [review]

**Review by chintogtokh — APPROVED** [review]
🔀

**Pull request merged**